internal/pkg/service/stream/sink/tablesink/storage/test/slice.go

```diff
@@ -1,20 +1,34 @@
 package test
 
 import (
+	"context"
+	"testing"
 	"time"
 
+	"github.com/benbjohnson/clock"
 	"github.com/c2h5oh/datasize"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
 
 	"github.com/keboola/keboola-as-code/internal/pkg/service/common/duration"
+	"github.com/keboola/keboola-as-code/internal/pkg/service/common/etcdop/op"
 	"github.com/keboola/keboola-as-code/internal/pkg/service/common/utctime"
 	"github.com/keboola/keboola-as-code/internal/pkg/service/stream/definition/column"
 	"github.com/keboola/keboola-as-code/internal/pkg/service/stream/sink/tablesink/storage"
 	"github.com/keboola/keboola-as-code/internal/pkg/service/stream/sink/tablesink/storage/compression"
 	"github.com/keboola/keboola-as-code/internal/pkg/service/stream/sink/tablesink/storage/level/local"
 	"github.com/keboola/keboola-as-code/internal/pkg/service/stream/sink/tablesink/storage/level/local/writer/disksync"
 	"github.com/keboola/keboola-as-code/internal/pkg/service/stream/sink/tablesink/storage/level/staging"
+	"github.com/keboola/keboola-as-code/internal/pkg/utils/errors"
 )
 
+// sliceRepository interface to prevent package import cycle.
+type sliceRepository interface {
+	Close(now time.Time, fileVolumeKey storage.FileVolumeKey) *op.AtomicOp[op.NoResult]
+	Get(k storage.SliceKey) op.WithResult[storage.Slice]
+	StateTransition(now time.Time, sliceKey storage.SliceKey, from, to storage.SliceState) *op.AtomicOp[storage.Slice]
+}
+
```

Comment:
I added these interface to prevent import cycles.

I want to have one `../storage/test` pkg with all helpers, not `testslice`,`testfile`, .... it would complicate imports/code.

Reply:
Go is very strict about this.

At first, the reason was that they wanted to simplify the implementation, 
then it turned out that it is actually super useful.

https://github.com/golang/go/issues/30247